src/main/java/com/github/lordrex34/config/model/ConfigFieldInfo.java

```diff
@@ -0,0 +1,193 @@
+import com.github.lordrex34.config.util.PropertiesParser;
+
+/**
+ * @author NB4L1 (original concept)
+ * @author lord_rex
+ */
+public final class ConfigFieldInfo
+{
+	private static final Logger LOGGER = LoggerFactory.getLogger(ConfigFieldInfo.class);
+	
+	/** The class that is being scanned. */
+	private final Class<?> _clazz;
+	
+	/** The field that contains {@link ConfigField} annotation. */
+	private final Field _field;
+	
+	/** The configuration field information annotation. */
+	private final ConfigField _configField;
+	
+	/** Group beginning marker annotation. */
+	private final ConfigGroupBeginning _beginningGroup;
+	
+	/** Group ending marker annotation. */
+	private final ConfigGroupEnding _endingGroup;
+	
+	/**
+	 * Constructs a new information container class for the field.
+	 * @param clazz the class that is being scanned
+	 * @param field the field that contains {@link ConfigField} annotation
+	 */
+	public ConfigFieldInfo(Class<?> clazz, Field field)
+	{
+		_clazz = clazz;
+		_field = field;
+		_configField = _field.getDeclaredAnnotation(ConfigField.class);
+		_beginningGroup = _field.getDeclaredAnnotation(ConfigGroupBeginning.class);
+		_endingGroup = _field.getDeclaredAnnotation(ConfigGroupEnding.class);
+	}
+	
+	/**
+	 * Loads and configures the field with its proper values.
+	 * @param configPath path of the properties file
+	 * @param properties the regular properties
+	 * @param overriddenProperties the content of {@code override.properties}
+	 */
+	public void load(Path configPath, PropertiesParser properties, PropertiesParser overriddenProperties)
+	{
+		// Skip inappropriate fields.
+		if (!Modifier.isStatic(_field.getModifiers()) || Modifier.isFinal(_field.getModifiers()))
+		{
+			LOGGER.debug("Skipping non static or final field: {}#{}", _clazz.getSimpleName(), _field.getName());
+			return;
+		}
+		
+		// If field is just a comment holder, then do not try to load it.
+		if (_configField.onlyComment())
+		{
+			return;
+		}
+		
+		try
+		{
+			final String propertyKey = _configField.name();
+			ConfigPropertyRegistry.add(_clazz.getPackage().getName(), configPath, propertyKey);
+			if (!_configField.reloadable() && ConfigManager.isReloading())
+			{
+				LOGGER.debug("Property '{}' retained with its previous value!", propertyKey);
+				return;
+			}
+			
+			// private field support
+			final boolean wasAccessible = _field.isAccessible();
+			try
+			{
+				if (!wasAccessible)
+				{
+					_field.setAccessible(true);
+				}
+				
+				_field.set(null, ConfigValueSuppliers.get(_configField.valueSupplier()).supply(_field, _configField, properties, overriddenProperties));
+				ConfigPostLoadHooks.get(_configField.postLoadHook()).load(properties, overriddenProperties);
+			}
+			finally
+			{
+				// restore field's visibility to the original
+				_field.setAccessible(wasAccessible);
+			}
+		}
+		catch (InstantiationException | IllegalAccessException e)
+		{
+			LOGGER.warn("Failed to set field!", e);
```

Comment:
you should explain me sometime what is the point of catching every random exception and letting the load go on... this should fail the shit out of loading ... that is why we have a nice dependency graph at the loading... the config cant be loaded -> nothing loads...

Reply:
Done.